### DISCUSSION TITLE
Re-instantiate Typedocs for `hidden` exported functions in `Account`

### DISCUSSION THREAD

**Discussion by maschad** [FuelLabs/fuels-ts#2429] [Ideas]
Since https://github.com/FuelLabs/fuels-ts/issues/835 was addressed, there are a number of functions within the [account](https://github.com/FuelLabs/fuels-ts/tree/master/packages/account) package which are relevant to consumers that are currently marked as `hidden` such as [getTransactionsSummaries](https://github.com/FuelLabs/fuels-ts/blob/1b896c9aeea8806510f95f7f68e17544a65ff9c8/packages/account/src/providers/transaction-summary/get-transaction-summary.ts#L140) for instance. 

Now that we have a more structured approach to documentation we can discuss how we want to articulate these exported functions. These include:

- [Transaction Summary methods](https://github.com/FuelLabs/fuels-ts/blob/master/packages/account/src/providers/transaction-summary/get-transaction-summary.ts)
- Some of the [operations methods](https://github.com/FuelLabs/fuels-ts/blob/master/packages/account/src/providers/transaction-summary/operations.ts)
- [Gas methods](https://github.com/FuelLabs/fuels-ts/blob/master/packages/account/src/providers/utils/gas.ts)

There may be some others missing, but I would like to hear others thoughts @FuelLabs/sdk-ts

**Comment by petertonysmith94** [comment]
Do functional methods even get populated in the Typedocs? I canny see any in the [documentation](https://docs.fuel.network/docs/fuels-ts/account/).

Types like should not be hidden (in typedocs):
- `CursorPaginationArgs` 
- `TransactionResult`

Functions like should be documented rather than just put in TypeDocs IMO:
- `getTransactionsSummaries`

I think as and when users ask for functionality that we have within the SDK, but isn't currently documented, we should make a better effort in rectifying this. That way the documentation can grow organically.

**Reply by maschad** [reply]
> Do functional methods even get populated in the Typedocs? I canny see any in the [documentation](https://docs.fuel.network/docs/fuels-ts/account/).

Maybe I'm misinterpreting what you are saying, but they are, [getTransactionSummary](https://docs.fuel.network/docs/fuels-ts/account/transactionresponse/#gettransactionsummary) is an example

> Functions like should be documented rather than just put in TypeDocs IMO:
> I think as and when users ask for functionality that we have within the SDK, but isn't currently documented, we should make a better effort in rectifying this. That way the documentation can grow organically.

I agree, I think we should also document these two particular methods in a guide as well, that was my intention. I suppose we can go about creating individual issues to document each folder. The purpose of the discussion was to nail down a clear scope of work for which `hidden` methods ought to be documented.

**Reply by petertonysmith94** [reply]
> Maybe I'm misinterpreting what you are saying, but they are, [getTransactionSummary](https://docs.fuel.network/docs/fuels-ts/account/transactionresponse/#gettransactionsummary) is an example

You interpreted right, just didn't see them in the [Table of Contents](https://docs.fuel.network/docs/fuels-ts/account/#module-fuel-tsaccount).

> The purpose of the discussion was to nail down a clear scope of work for which hidden methods ought to be documented.

If that is the purpose, would it make sense to list out all the methods/types that we propose to make un-hidden? (in the description)

**Reply by maschad** [reply]
Correct,  I began the list in the description, so all of the [Transaction Summary methods](https://github.com/FuelLabs/fuels-ts/blob/master/packages/account/src/providers/transaction-summary/get-transaction-summary.ts) and all the [Gas methods](https://github.com/FuelLabs/fuels-ts/blob/master/packages/account/src/providers/utils/gas.ts) I'm not sure if this is comprehensive though.

> Some of the [operations methods](https://github.com/FuelLabs/fuels-ts/blob/master/packages/account/src/providers/transaction-summary/operations.ts)

I should have been explicit here, I am referring to:

- `getReceiptsByType`
- `getTransactionTypeName`
- `getWithdrawFromFuelOperations`
- `extractTransferOperationFromReceipt`

Are there any others you think would be appropriate?

**Reply by Torres-ssf** [reply]
The list should include `getTransactionsSummaries`

**Comment by maschad** [comment]
Closing this discussion for now as I've created [this issue](https://github.com/FuelLabs/fuels-ts/issues/3002) based on the feedback.
